src/eei.cpp

```diff
@@ -206,6 +218,10 @@ namespace HeraVM {
 
       HERA_DEBUG << "callDataCopy " << hex << resultOffset << " " << dataOffset << " " << length << dec << "\n";
 
+      heraAssert(ffs(GasSchedule::copy) + (ffs(length) - 5) <= 64, "Gas charge overflow");
+      heraAssert(numeric_limits<uint64_t>::max() - GasSchedule::verylow >= GasSchedule::copy * ((static_cast<uint64_t>(length) + 31) / 32), "Gas charge overflow");
+      takeGas(GasSchedule::verylow + GasSchedule::copy * ((static_cast<uint64_t>(length) + 31) / 32));
```

Comment:
I really think we should have a helper for this since there 3 copies now.

Reply:
Could open an issue for this and do it in a separate PR, possibly a refactoring pass.
Might as well get it working now.